de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/impl/OrderLineBL.java

```diff
@@ -956,30 +961,69 @@ public void copyOrderLineCounter(final org.compiere.model.I_C_OrderLine line, fi
 	public void updateNoPriceConditionsColor(final I_C_OrderLine orderLine)
 	{
 
-		final String colorName = Services.get(ISysConfigBL.class).getValue(SYSCONFIG_NoPriceConditionsColorName, "-");
+		final String colorName = getNoPricingConditionsColorName();
 
 		final int discountSchemaBreakId = orderLine.getM_DiscountSchemaBreak_ID();
 
 		if (discountSchemaBreakId > 0)
 		{
 			// the discountSchemaBreak was eventually set. The color warning is no longer needed
 			orderLine.setNoPriceConditionsColor_ID(-1);
-			
+
 			return;
 		}
 
 		final int colorId = getNoPriceConditionsColorId(colorName);
-		
+
 		if (colorId > 0)
 		{
 			orderLine.setNoPriceConditionsColor_ID(colorId);
 		}
 
 	}
 
+	private String getNoPricingConditionsColorName()
+	{
+		return Services.get(ISysConfigBL.class).getValue(SYSCONFIG_NoPriceConditionsColorName, "-");
```

Comment:
IMHO, u can embed this in getNoPriceConditionsColorId.
Also there is no point for `getNoPriceConditionsColorId` to have a `String name` parameter.

Reply:
also, get rid of `getNoPricingConditionsColorName`